crates/core/src/execution_context.rs

```diff
@@ -177,6 +177,18 @@ impl From<&ReducerContext> for txdata::Inputs {
     }
 }
 
+impl Clone for ExecutionContext {
```

Comment:
A comment for why we need a custom `Clone`?

Reply:
Sure I 'll add, 
FYI, it's there so that `ctx` can be shared across thread without lock on metrics fields, which might be mutated.